app/services/sso/openid_connect.rb

```diff
@@ -4,27 +4,48 @@ class OpenidConnect < Base
     attr_reader :current_user
 
     def available?
+      if session[:sso_method] == "SSO::OpenidConnect"
+        session.delete(:sso_method) unless session[:user]
+        return true
+      end
       ((controller.api_request? && bearer_token_set?) || http_token.present?) && valid_issuer?
     end
 
     def authenticate!
+      self.has_rendered = true
+      controller.redirect_to login_url
+    end
+
+    def authenticated?
       payload = jwt_token.decode
-      return nil if payload.nil?
+      return false if payload.nil?
       user = find_or_create_user_from_jwt(payload)
       @current_user = user
       update_session(payload)
       user&.login
-    end
-
-    def authenticated?
-      self.user = User.current.presence || authenticate!
+      self.user = User.current.presence
```

Comment:
same for `user` and `self.user` - afaict they aren't used outside this function

Reply:
I am not sure if we must keep the `self.user` or not. It is being used in the base class, that opinid_connect.rb inherits from. Check this out: https://github.com/theforeman/foreman/blob/develop/app/services/sso/base.rb#L4

Let me know if we will need it for the safer side or not.